collect_app/src/main/java/org/odk/collect/android/utilities/MediaManager.java

```diff
@@ -24,7 +24,7 @@
  * remember an original media file answer (no matter how many times this answer is replaced), in order
  * to be able to restore the original answer in case of ignoring changes.
  */
-public enum MediaManager {
+public enum MediaManager implements QuestionMediaManager {
```

Comment:
I don't think this is quite right after a second look. As much as want to avoid static singletons. If there is some state that we want to keep shared at an application level (i.e. between multiple activities or services) then we should use Dagger to setup and inject a singleton (using `@Singleton`).  

In this case I don't believe the state needs to persist outside of form entry so it really only needs to last as long as the Activity.  This means we could confuse ourselves by ending up with state hanging around between form entry "sessions". A good way to share state like this would be to use a view model where the `ViewProvider` uses the activity as its owner. Does that make sense as a way forward?

Reply:
If so, it might be good to discuss the plans/approach here so we can come up with something that makes sense as it's a little awkward I think!

Reply:
@seadowg If I use the `ViewModel` then I would be creating it using the widget context, and it will be there for all the widgets on the screen?

Reply:
I think if that's simpler to do right now that's a good plan. Maybe we can move the logic to a viewmodel in the future.